resource_control: add set resource group and resource group hint (#12866)
pingcap/docs: sql-statements/sql-statement-set-resource-group.md

---

# SET RESOURCE GROUP

`SET RESOURCE GROUP` is used to set the resource group for the current user session.

## Synopsis

**SetResourceGroupStmt:**

```ebnf+diagram
SetResourceGroupStmt:
   "SET" "RESOURCE" "GROUP" ResourceGroupName
ResourceGroupName:
   Identifier
```

## Examples

Create a user `user1`, create two resource groups `rg1` and `rg2`, and bind the user `user1` to the resource group `rg1`.

```sql
CREATE USER 'user1';
CREATE RESOURCE GROUP 'rg1' RU_PER_SEC = 1000;
ALTER USER 'u1' RESOURCE GROUP `rg1`;
```

Use `user1` to log in and view the resource group bound to the current user.

```sql
SELECT CURRENT_RESOURCE_GROUP();
```

```
+--------------------------+
| CURRENT_RESOURCE_GROUP() |
+--------------------------+
| rg1                      |
+--------------------------+
1 row in set (0.00 sec)
```

Execute `SET RESOURCE GROUP` to set the resource group for the current session to `rg2`.

```sql
SET RESOURCE GROUP `rg2`;
SELECT CURRENT_RESOURCE_GROUP();
```

```
+--------------------------+
| CURRENT_RESOURCE_GROUP() |
+--------------------------+
| rg2                      |
+--------------------------+
1 row in set (0.00 sec)
```

> Execute `SET RESOURCE GROUP` to set the resource group of the current session as the default resource group.

Author: ```suggestion
Execute `SET RESOURCE GROUP` to specify the current session to use the default resource group.
```
